src/wp-includes/post.php

```diff
@@ -8018,3 +8019,38 @@ function use_block_editor_for_post_type( $post_type ) {
 	 */
 	return apply_filters( 'use_block_editor_for_post_type', true, $post_type );
 }
+
+/**
+ * Adds sync_status meta field to the wp_block post type so an unsynced option can be added.
+ *
+ * @since 6.3.0
+ *
+ * @see https://github.com/WordPress/gutenberg/pull/51144
+ *
+ * @return void
+ */
+function wp_register_wp_block_postmeta() {
```

Comment:
```suggestion
function wp_register_initial_post_meta() {
```

WP registers all the post types in one function (along with statuses), `create_initial_post_types`, so I'm leaning towards using a generic function name to allow for additional meta data to be registered in the future. 

Obviously if this changes then the equivalent is needed in default-actions.php -- would this cause problems in GB?

Reply:
That makes sense. Done. I am not sure how it affects GB as a different method name there anyway, but have made a similar name change there just so people can at least map the `gutenberg_` prefixed method to this one if needed.

Reply:
Do we want to stick with the `create_initial` naming pattern? We did most recently in `create_initial_theme_features`.

Reply:
Yeah, that makes sense to me. Sorry for the bad steer, Glen.

Reply:
no worries, have changed to `_create_initial_`

Reply:
> do you know if we need to update the GB gutenberg_wp_block_register_post_meta to gutenberg_register_initial_post_meta to match this change

I don't _think_ so, but we'll need to preserve the existing code in GB in the compat folder for use with previous versions of WP.